libraries/aws_route_table_association.rb

```diff
@@ -0,0 +1,57 @@
+class AWSRouteTableAssociation < Inspec.resource(1)
+  name 'aws_route_table_association'
+  desc 'Test the configuration of a class AWS Route Table Association'
+  example "
+    describe aws_vpc_endpoint('rtb-1234', 'rtbassoc-1234') do
+      it { should exist }
+      it { should be_main }
+      its('id') { should eq 'rtbassoc-1234' }
+      its('subnet_id') { should eq 'subnet-1234' }
+      its('route_table_id') { should eq 'rtb-1234' }
+    end
+  "
+
+  def initialize(route_table_id, route_table_association_id)
```

Comment:
Is the route table ID really needed if you actually already have the route table association ID that you want to test?

I guess you could say that you are iterating over all the route tables and their associations and find the content of the association only if you find something that matches. It may be slower that requesting directly the corresponding route ID but at the same time, I'm not sure this is the kind of resources you expect to have thousands of them?

Reply:
The thing is that you can only get the route table associations if you have the route table id by iterating through the associations. I did not find a way of getting directly the route table association using the aws sdk. If you know of something, that would be great.

Reply:
Yeah, I got that, my point was: it's apparently possible to describe "all the route tables" (without specifying a route table ID), from each you could loop over and then loop over their associations, something like:

```ruby
route_tables =  ec2.describe_route_tables().route_tables
route_tables.each do |route_table|
  route_table.associations.each do |assoc|
    if assoc['route_table_association_id'] == route_table_association_id
      # found it, break and bla
    end
  # for-else: no route table association ID found
end
```

I don't know if you should expect to have a lot of route tables anyway, so I think it would be fine and the interface of the resource would be simpler.

(and in case it's *really* needed, I guess we can add the *route table ID* as an optional argument, but I don't think it's really useful right now.)

Reply:
ah, got it now. Yeah that works, will make the change.